component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -65,24 +66,35 @@ protected void initiateAuthenticationRequest(HttpServletRequest httpServletReque
             throws AuthenticationFailedException {
         String authEndpoint = null;
         try {
-            if (getAuthenticatorConfig().getParameterMap()
-                    .get(X509CertificateConstants.AUTHENTICATION_ENDPOINT) != null) {
-                authEndpoint = getAuthenticatorConfig().getParameterMap()
-                        .get(X509CertificateConstants.AUTHENTICATION_ENDPOINT);
+            if (authenticationContext.isRetrying()) {
+                if (Boolean.parseBoolean(String.valueOf(authenticationContext.
+                        getProperty(X509CertificateConstants.UNABLE_TO_FIND_CERTIFICATE)))) {
+                    String errorPage = IdentityUtil.getServerURL(X509CertificateConstants.ERROR_PAGE, false, false);
+                    String errorPageUrl =
+                            errorPage + ("?sessionDataKey=" + authenticationContext.getContextIdentifier()) +
+                                    "&authenticators=" + getName() + X509CertificateConstants.RETRY_PARAM;
+                    httpServletResponse.sendRedirect(errorPageUrl);
+                }
             } else {
-                authEndpoint = X509CertificateConstants.AUTH_ENDPOINT;
-            }
-            String queryParams = FrameworkUtils.getQueryStringWithFrameworkContextId(
-                    authenticationContext.getQueryParams(), authenticationContext.getCallerSessionKey(),
-                    authenticationContext.getContextIdentifier());
-            String encodedUrl = (authEndpoint + ("?" + queryParams));
-            httpServletResponse.sendRedirect(encodedUrl);
-            if (log.isDebugEnabled()) {
-                log.debug("Request send to " + authEndpoint);
+                if (getAuthenticatorConfig().getParameterMap()
+                        .get(X509CertificateConstants.AUTHENTICATION_ENDPOINT) != null) {
+                    authEndpoint = getAuthenticatorConfig().getParameterMap()
+                            .get(X509CertificateConstants.AUTHENTICATION_ENDPOINT);
+                } else {
+                    authEndpoint = X509CertificateConstants.AUTH_ENDPOINT;
```

Comment:
Define the constant value for authEndpoint at the top and change the value if the property X509CertificateConstants.AUTHENTICATION_ENDPOINT has the value in authenticator config.